### PR TITLE
Mohammad/Reset-password to TS

### PR DESCRIPTION
**Pull request by mohammad-rework**
Changes:

-   Convert Reset-password page to TS

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [x] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/pages/reset-password/index.tsx [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+            ),
+        })
+        binary_socket.close()
+    }
+}
+
+const ResetPassword: React.ReactNode = () => {
+    const initialValues: EmailType = { email: '' }
+    return (
+        <Layout type="static" margin_top={'0'}>
```
Comment (yashim-deriv) [inline src/pages/reset-password/index.tsx:81]:
```suggestion
        <Layout type="static" margin_top='0'>
```

File: src/pages/reset-password/index.tsx [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+                                    error={errors.email}
+                                    value={trimSpaces(values.email)}
+                                    handleError={resetForm}
+                                    onChange={handleChange}
+                                    onBlur={handleBlur}
+                                    autoComplete="off"
+                                    type="text"
+                                    label={localize('Email')}
+                                    background="white"
+                                    placeholder={'example@email.com'}
```
Comment (yashim-deriv) [inline src/pages/reset-password/index.tsx:132]:
```suggestion
                                    placeholder='example@email.com'
```

File: src/pages/reset-password/index.tsx [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+    const email_error = validation.required(email) || validation.email(email)
+
+    if (email_error) {
+        errors.email = email_error
+    }
+
+    return errors
+}
+
+const resetSubmission = (values, actions) => {
```
Comment (yashim-deriv) [inline src/pages/reset-password/index.tsx:50]:
```suggestion
const resetSubmission = (values, actions) => {
```
Don't we need to type this?

File: src/pages/reset-password/index.tsx [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+        actions.setStatus({
+            success: localize(
+                'Please check your email and click on the link provided to reset your password.',
+            ),
+        })
+        binary_socket.close()
+    }
+}
+
+const ResetPassword: React.ReactNode = () => {
```
Comment (yashim-deriv) [inline src/pages/reset-password/index.tsx:78]:
```suggestion
const ResetPassword = () => {
```
Based on our latest TS standard, we can omit return types of React Components

File: src/pages/reset-password/index.tsx [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+import { Header, Text } from 'components/elements'
+import { Input, Button } from 'components/form'
+import validation from 'common/validation'
+import { trimSpaces } from 'common/utility'
+import { BinarySocketBase } from 'common/websocket/socket_base'
+import Login from 'common/login'
+
+type EmailType = { email: string }
+
+type ErrorType = { email?: string }
```
Comment (prince-deriv) [inline src/pages/reset-password/index.tsx:16]: You can use Utility types to manage/reuse declared types to fit on specifications. 

`
type ErrorType = Partial<EmailType>
`

**Review by habib-deriv — APPROVED** [review]

**Pull request merged**